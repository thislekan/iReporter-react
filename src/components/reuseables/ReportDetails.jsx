import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import cx from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import style from '../../styles/ReportDetails.css';

function ReportDetails(props) {
  const { incident, mapUrl } = props;
  return (
    <div className={style['report-body-wrapper']}>
      <div id="detailed-incident">
        <div className={cx(style['report-body-wrapper'], style.pushed)}>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>ID: </h4>
            </div>
            <p id="incident-id">{incident.id}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Type: </h4>
            </div>
            <p id="incident-type">{incident.type}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Created: </h4>
            </div>
            <p id="incident-date">{moment(Number(incident.createdOn)).format('MMM Do YY')}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Status: </h4>
            </div>
            <p id="incident-status">{incident.status}</p>
          </div>
          <div className={style['report-details']}>
            <div className={style['incident-intro']}>
              <FontAwesomeIcon icon='check-circle' />
              <h4>Title: </h4>
            </div>
            <p id="incident-title">{incident.title}</p>
          </div>
          <div className={style['location-div']}>
            <div className={style['report-details']}>
              <div className={style['incident-intro']}>
                <FontAwesomeIcon icon='check-circle' />
                <h4>Location: </h4>
              </div>
              <p id="incident-location">{incident.location}</p>
            </div>
            <div className={style.map}>
              <h4>Map</h4>
              <div className={style['map-region']}>
                <iframe
                  src={mapUrl}
                  frameBorder="0"
                  width="100%"
                  height="100%"
                  title={`${incident.location}'s map`}
                  style={{ border: 0, position: 'absolute' }}
                  allowFullScreen
                />
              </div>
            </div>
            <div className={style['comment-div']}>
              <h4>Comment: </h4>
              <textarea
                name="comment"
                id="incident-comment"
                disabled
                value={(incident.comment) ? incident.comment : ''}
              />
            </div>
          </div>
          {(incident.images || incident.video) && <div className={style['image-and-video-div']}>
            <h4>Images or Videos: </h4>
            <div className={style['content-wrapper']}>
              <div id="incident-video-div"></div>
              {incident.images && incident.images.length >= 1 && (
                <div className={style['image-container']} id="incident-img-div">
                  {incident.images.map((image, index) => <img src={image} key={index} alt="" />)}
                </div>
              )}
            </div>
          </div>}
        </div>
      </div>
    </div>
  );
}

ReportDetails.propTypes = {
  incident: PropTypes.object.isRequired,
  mapUrl: PropTypes.string,
};

ReportDetails.defaultProps = {
  mapUrl: '',
};

export default ReportDetails;
