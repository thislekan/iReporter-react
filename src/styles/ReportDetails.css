.report-details{
    padding: .6rem .75rem;
    display: grid;
    grid-template-columns: 10rem auto;
    align-items: center;
    justify-content: space-between;
}
.report-details p{
    margin: .5rem 0;
    margin-left: 1.5rem;
    text-align: right;
}
.incident-intro{
    display: flex;
    align-items: center;
}
.incident-intro svg{
    color: #117d5a;
}
.map-region{
    height: 20rem;
    border: 1px solid;
    position: relative;
}
.map{
    margin: 2rem 0;
}
.content-wrapper{
    height: 100%;
    margin: 1rem 0;
}
.image-container{
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: auto;
    grid-gap: 1rem;
    margin-top: 1rem;
}
.image-container img{
    height: 14.5rem;
    width: 100%;
    object-fit: scale-down;
    object-position: center;
    border: 1px solid rgba(29, 57, 74, 0.35);
    border-radius: .25rem;
}
@media only screen and (min-width: 700px) {
  .map-region{
        height: 30rem;
    }
    .image-container img{
        height: 12rem;
    }
}
@media only screen and (min-width: 900px){
  .image-container img{
        height: 22rem;
    }
}