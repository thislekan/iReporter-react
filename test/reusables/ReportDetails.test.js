import React from 'react';
import { shallow } from 'enzyme';
import ReportDetails from '../../src/components/reuseables/ReportDetails.jsx';

const mock = {
  isLoading: false,
  message: '',
  id: 2,
  createdBy: '',
  createdOn: '',
  creator: '',
  updatedOn: '',
  title: '',
  type: '',
  location: '',
  status: '',
  comment: 'jhdfhfdjk',
  images: ['hi', 'helo'],
  videos: ['fdkfdn'],
};

describe('<ReportDetails />', () => {
  it('should render ReportDetails', () => {
    const wrapper = shallow(
      <ReportDetails
        incident={mock}
        mapUrl="youth"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a different content when the route is /admin', () => {
    const mimic = mock;
    mimic.images = '';
    mimic.videos = '';
    mimic.comment = '';
    const wrapper = shallow(
      <ReportDetails
        incident={mimic}
        mapUrl="youth"
      />,
    );
    const textArea = wrapper.find('#incident-comment');
    expect(textArea.props().value).toBe('');
    expect(wrapper).toMatchSnapshot();
  });
});
