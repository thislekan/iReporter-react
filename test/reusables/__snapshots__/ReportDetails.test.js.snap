// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportDetails /> should render ReportDetails 1`] = `
<div>
  <div
    id="detailed-incident"
  >
    <div
      className=""
    >
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Title: 
          </h4>
        </div>
        <p
          id="incident-title"
        />
      </div>
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Type: 
          </h4>
        </div>
        <p
          id="incident-type"
        />
      </div>
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Created: 
          </h4>
        </div>
        <p>
           
        </p>
      </div>
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Status: 
          </h4>
        </div>
        <p
          id="incident-status"
        />
      </div>
      <div>
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Location: 
            </h4>
          </div>
          <p
            id="incident-location"
          />
        </div>
        <div>
          <h4>
            Map
          </h4>
          <div>
            <iframe
              allowFullScreen={true}
              frameBorder="0"
              height="100%"
              src="youth"
              style={
                Object {
                  "border": 0,
                  "position": "absolute",
                }
              }
              title="'s map"
              width="100%"
            />
          </div>
        </div>
        <div>
          <h4>
            Comment: 
          </h4>
          <textarea
            disabled={true}
            id="incident-comment"
            name="comment"
            value="jhdfhfdjk"
          />
        </div>
      </div>
      <div>
        <h4>
          Images or Videos: 
        </h4>
        <div>
          <div
            id="incident-video-div"
          />
          <div
            id="incident-img-div"
          >
            <img
              alt=""
              key="0"
              src="hi"
            />
            <img
              alt=""
              key="1"
              src="helo"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ReportDetails /> should render a different content when the route is /admin 1`] = `
<div>
  <div
    id="detailed-incident"
  >
    <div
      className=""
    >
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Title: 
          </h4>
        </div>
        <p
          id="incident-title"
        />
      </div>
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Type: 
          </h4>
        </div>
        <p
          id="incident-type"
        />
      </div>
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Created: 
          </h4>
        </div>
        <p>
           
        </p>
      </div>
      <div>
        <div>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="check-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          <h4>
            Status: 
          </h4>
        </div>
        <p
          id="incident-status"
        />
      </div>
      <div>
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Location: 
            </h4>
          </div>
          <p
            id="incident-location"
          />
        </div>
        <div>
          <h4>
            Map
          </h4>
          <div>
            <iframe
              allowFullScreen={true}
              frameBorder="0"
              height="100%"
              src="youth"
              style={
                Object {
                  "border": 0,
                  "position": "absolute",
                }
              }
              title="'s map"
              width="100%"
            />
          </div>
        </div>
        <div>
          <h4>
            Comment: 
          </h4>
          <textarea
            disabled={true}
            id="incident-comment"
            name="comment"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
